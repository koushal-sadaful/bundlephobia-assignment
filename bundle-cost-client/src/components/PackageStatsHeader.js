import {Button, Card, Icon} from "semantic-ui-react";
import React from "react";
import {Link} from "react-router-dom";


const PackageStatsHeader = (props) => {
    return (
        <div className="package-stats-header">
            <Card fluid>
                <Card.Content>
                    <Card.Header style={{
                        fontSize: '2.5em',
                        padding: '0.3em'
                    }}>
                        {props.name || "Not Available"}
                    </Card.Header>
                    <Card.Meta>
                        {`[${props.version || ""}]`}
                    </Card.Meta>
                    <Card.Description>
                        {props.description || ""}
                    </Card.Description>
                </Card.Content>
                <Card.Content extra>
                    <Link to="/">
                        <Button icon labelPosition='left'>
                            <Icon name='search'/>
                            Search another package
                        </Button>
                    </Link>
                </Card.Content>
            </Card>
        </div>
    );
};

export default PackageStatsHeader