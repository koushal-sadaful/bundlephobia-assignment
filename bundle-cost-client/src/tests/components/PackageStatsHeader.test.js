import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {configure, mount} from 'enzyme';
import PackageStatsHeader from "../../components/PackageStatsHeader";
import {Card, Button} from "semantic-ui-react";
import {BrowserRouter} from "react-router-dom";
import {runTheSnapshotTest} from "../testUtils/TestHelpers";

configure({adapter: new Adapter()});

describe("PackageStatsHeader", () => {
    let wrapper = null;
    let defaultTestProps = {
        name: "De Brie",
        version: "1.0",
        description: "Le fromage que j'aime"
    };

    afterEach(() => {
        if (wrapper){
            wrapper.unmount()
            wrapper = null
        }
    });

    it("should render package details", () => {
        wrapper = mount(<BrowserRouter><PackageStatsHeader {...defaultTestProps} /> </BrowserRouter>);

        expect(wrapper.find(Card.Header).props().children).toMatch(defaultTestProps.name);
        expect(wrapper.find(Card.Meta).props().children).toMatch(`[${defaultTestProps.version}]`);
        expect(wrapper.find(Card.Description).props().children).toMatch(defaultTestProps.description);
        expect(wrapper.find(Button).props().children[1]).toMatch("Search another package");
    });

    it("should render nothing if missing props", () => {
        const testProps = {
            name: "De Brie"
        };
        wrapper = mount(<BrowserRouter> <PackageStatsHeader {...testProps} /> </BrowserRouter>);

        expect(wrapper.find(Card.Header).props().children).toMatch(testProps.name);
        expect(wrapper.find(Card.Meta).props().children).toMatch(`[]`);
        expect(wrapper.find(Card.Description).props().children).toMatch("");
    });

    it("should pass the snapshot test", () => {
        runTheSnapshotTest(<BrowserRouter> <PackageStatsHeader {...defaultTestProps} /> </BrowserRouter>)
    });

});