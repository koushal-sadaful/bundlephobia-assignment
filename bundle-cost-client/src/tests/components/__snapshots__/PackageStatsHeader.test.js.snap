// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackageStatsHeader should pass the snapshot test 1`] = `
Array [
  " ",
  <div
    className="package-stats-header"
  >
    <div
      className="ui fluid card"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
          style={
            Object {
              "fontSize": "2.5em",
              "padding": "0.3em",
            }
          }
        >
          De Brie
        </div>
        <div
          className="meta"
        >
          [1.0]
        </div>
        <div
          className="description"
        >
          Le fromage que j'aime
        </div>
      </div>
      <div
        className="extra content"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <button
            className="ui icon left labeled button"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="search icon"
              onClick={[Function]}
            />
            Search another package
          </button>
        </a>
      </div>
    </div>
  </div>,
  " ",
]
`;
